assembl/static2/js/app/components/debate/survey/post.jsx

```diff
@@ -0,0 +1,177 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { graphql, compose } from 'react-apollo';
+import gql from 'graphql-tag';
+import { Translate } from 'react-redux-i18n';
+import { getConnectedUserId } from '../../../utils/globalFunctions';
+import { getIfPhaseCompletedByIdentifier } from '../../../utils/timeline';
+import PostCreator from './postCreator';
+import Circle from '../../svg/circle';
+import Like from '../../svg/like';
+import Disagree from '../../svg/disagree';
+
+class Post extends React.Component {
+  constructor(props) {
+    super(props);
+    const { post } = this.props;
+    this.state = {
+      like: post.sentimentCounts.like,
+      disagree: post.sentimentCounts.disagree,
+      mySentiment: post.mySentiment
+    };
+    this.handleSentiment = this.handleSentiment.bind(this);
+  }
+  handleSentiment(event, type) {
+    const isUserConnected = getConnectedUserId() !== null;
+    const { redirectToLogin } = this.props;
+    if (isUserConnected) {
+      const { debateData } = this.props.debate;
+      const isPhaseCompleted = getIfPhaseCompletedByIdentifier(debateData.timeline, 'survey');
+      if (!isPhaseCompleted) {
+        const target = event.currentTarget;
+        const isMySentiment = this.state.mySentiment === type;
+        if (isMySentiment) {
+          this.deleteSentiment(target);
+        } else {
+          this.addSentiment(target, type);
+        }
+      }
+    } else {
+      redirectToLogin();
+    }
+  }
+  addSentiment(target, type) {
+    const { id } = this.props;
+    this.props.addSentiment({ variables: { postId: id, type: type } })
```

Comment:
Yes, in apollo, you use the mutate api, it triggers behind the scene an action, and take care of updating the redux store via the apollo reducer.